src/components/Variables/VariableHistoryTable.tsx

```diff
@@ -75,179 +75,184 @@ export const VariableHistoryTable = ({ projectId, variableId }: VariableHistoryT
             change: {
-              field: 'value',
-              previous: previousValue?.numberValue.toString() ?? '',
-              current: currentValue?.numberValue.toString() ?? '',
+              field: 'status',
+              previous: previous?.status ?? '',
+              current: history.status,
             },
           });
         }
-      } else if (variable.type === 'Text') {
-        const currentValue = history.variableValues[0] as VariableValueTextValue | undefined;
-        const previousValue = previous?.variableValues[0] as VariableValueTextValue | undefined;
 
-        if (previousValue?.textValue !== currentValue?.textValue) {
-          newRows.push({
-            date: history.createdTime,
-            editedBy: history.createdBy,
-            change: {
-              field: 'value',
-              previous: previousValue?.textValue ?? '',
-              current: currentValue?.textValue ?? '',
-            },
-          });
-        }
-      } else if (variable.type === 'Select') {
-        const currentValues = history.variableValues[0] as VariableValueSelectValue | undefined;
-        const previousValues = previous?.variableValues[0] as VariableValueSelectValue | undefined;
+        if (variable.type === 'Table') {
+          const currentTableValueIds = history.variableValues
+            .toSorted((a, b) => a.listPosition - b.listPosition)
+            .map((value) => value.id)
+            .join(',');
 
-        const currentOptionValues = (currentValues?.optionValues ?? [])
-          .toSorted()
-          .map((id) => variable.options.find((option) => option.id === id))
-          .filter((option): option is SelectOptionPayload => option !== undefined) // drop unrecognized select options
-          .map((option) => option.name)
-          .join(',');
+          const previousTableValueIds = (previous?.variableValues ?? [])
+            .toSorted((a, b) => a.listPosition - b.listPosition)
+            .map((value) => value.id)
+            .join(',');
 
-        const previousOptionValues = (previousValues?.optionValues ?? [])
-          .toSorted()
-          .map((id) => variable.options.find((option) => option.id === id))
-          .filter((option): option is SelectOptionPayload => option !== undefined)
-          .map((option) => option.name)
-          .join(',');
+          if (previousTableValueIds !== currentTableValueIds) {
+            newRows.push({
+              date: history.createdTime,
+              editedBy: history.createdBy,
+              change: {
+                field: 'table',
+                previous: previousTableValueIds,
+                current: currentTableValueIds,
+              },
+            });
+          }
+        } else if (variable.type === 'Date') {
+          const currentValue = history.variableValues[0] as VariableValueDateValue | undefined;
+          const previousValue = previous?.variableValues[0] as VariableValueDateValue | undefined;
```

Comment:
Can these be pulled up in scope since they are used in almost every variable type?

Reply:
Disregard, I see now this stuff was just indented.